resilience4j-timelimiter/src/main/java/io/github/resilience4j/timelimiter/internal/TimeLimiterImpl.java

```diff
@@ -55,6 +55,37 @@ public TimeLimiterImpl(String name, TimeLimiterConfig timeLimiterConfig) {
         };
     }
 
+    @Override
+    public <T, F extends CompletionStage<T>> Supplier<CompletionStage<T>> decorateCompletionStage(Supplier<F> supplier) {
+        CompletableFuture<T> future = supplier.get().toCompletableFuture();
+        ScheduledFuture<?> timeoutFuture =
```

Comment:
The ScheduledFuture should not be created when `decorateCompletionStage` is invoked, but when the returned, decorated Supplier is invoked.